<a href="https://colab.research.google.com/github/CppR-Repo/TT-2C2025-Data-Analitycs-Notebooks/blob/main/PreEntrega/MiPre-entrega/Pre-entrega.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Análisis de datos con Pyhon - Pre-entrega
Nombre y apellido: Cristian Pablo Portillo Rios

# Etapa 1

## Script básico
Actividades 2 y 3

In [1]:
# Generamos una lista de diccionarios que contenga algunos datos
ventas = [
  {'producto': 'casablanca', 'tipo': 'latex', 'descripcion': 20, 'precio': 120000.0, 'cantidad': 2},
  {'producto': 'satinol', 'tipo': 'sintetico', 'descripcion': 4, 'precio': 75000.0, 'cantidad': 3},
  {'producto': 'andina', 'tipo': 'fijador', 'descripcion': 1, 'precio': 7800.0, 'cantidad': 2},
  {'producto': 'comodin', 'tipo': 'enduido', 'descripcion': 10, 'precio': 20500.0, 'cantidad': 1},
  {'producto': 'aa', 'tipo': 'lija', 'descripcion': 100, 'precio': 2100.0, 'cantidad': 10}
]

In [2]:
# Calculamos las ventas acumuladas - map
# 1. Calcula la venta de cada producto con map
ventas_individuales = map(lambda x:x['precio'] * x['cantidad'], ventas)

# 2. Sumar todas las ventas individuales con sum
ventas_acumuladas = sum(ventas_individuales)
print(f"Las ventas acumuladas son: {ventas_acumuladas}")

Las ventas acumuladas son: 522100.0


In [3]:
# Calculamos las ventas acumuladas - listas por comprensión
venta_acumuladas = sum([venta["precio"] * venta["cantidad"] for venta in ventas])
venta_acumuladas

522100.0

## Import Datasets
Actividad 1

In [4]:
import pandas as pd

In [5]:
# Import dataset ventas
df_ventas = pd.read_csv("https://raw.githubusercontent.com/CppR-Repo/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/PreEntrega/Datasets/ventas.csv")

In [6]:
# Import dataset clientes
df_clientes = pd.read_csv("https://raw.githubusercontent.com/CppR-Repo/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/PreEntrega/Datasets/clientes.csv")

In [7]:
# Import dataset marketing
df_marketing = pd.read_csv("https://raw.githubusercontent.com/CppR-Repo/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/PreEntrega/Datasets/marketing.csv")

In [8]:
df_ventas

id_venta                producto   precio  cantidad fecha_venta  \
0          792       Cuadro decorativo   $69.94       5.0  02/01/2024   
1          811         Lámpara de mesa  $105.10       5.0  02/01/2024   
2         1156                Secadora   $97.96       3.0  02/01/2024   
3         1372                Heladera  $114.35       8.0  02/01/2024   
4         1546                Secadora  $106.21       4.0  02/01/2024   
...        ...                     ...      ...       ...         ...   
3030      1837         Horno eléctrico  $104.12       9.0  30/12/2024   
3031      2276                  Laptop   $85.27       9.0  30/12/2024   
3032      2696                  Laptop  $107.81       4.0  30/12/2024   
3033      2913              Smartphone   $99.85       7.0  30/12/2024   
3034      2930  Consola de videojuegos   $55.47       6.0  30/12/2024   

              categoria  
0            Decoración  
1            Decoración  
2     Electrodomésticos  
3     Electrodomésticos  
4     Electrodomésticos  
...                 ...  
3030  Electrodomésticos  
3031        Electrónica  
3032        Electrónica  
3033        Electrónica  
3034        Electrónica  

[3035 rows x 6 columns]

## Dataframe ventas

### Análisis exploratorio de **datos**
Actividad 4

Aplicamos los métodos de EDA vistos en clase, `info`, `columns`, `dtypes`, `head`, `tail`, `sample`, `unique`, `value_counts`, `describe`, `duplicated`, `isnull`, etc. y documentamos las observaciones:

In [9]:
# Aplicamos info() para familiarizarnos con el dataframe
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3035 entries, 0 to 3034
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id_venta     3035 non-null   int64  
 1   producto     3035 non-null   object 
 2   precio       3033 non-null   object 
 3   cantidad     3033 non-null   float64
 4   fecha_venta  3035 non-null   object 
 5   categoria    3035 non-null   object 
dtypes: float64(1), int64(1), object(4)
memory usage: 142.4+ KB


In [10]:
# Visualizamos los primeros registros con head()
# también podemos usar tail() para visualizar los últimos
df_ventas.head()

id_venta           producto   precio  cantidad fecha_venta  \
0       792  Cuadro decorativo   $69.94       5.0  02/01/2024   
1       811    Lámpara de mesa  $105.10       5.0  02/01/2024   
2      1156           Secadora   $97.96       3.0  02/01/2024   
3      1372           Heladera  $114.35       8.0  02/01/2024   
4      1546           Secadora  $106.21       4.0  02/01/2024   

           categoria  
0         Decoración  
1         Decoración  
2  Electrodomésticos  
3  Electrodomésticos  
4  Electrodomésticos

In [11]:
# Usar sample() para tomar registros al azar
df_ventas.sample(5)

id_venta    producto   precio  cantidad fecha_venta          categoria
1452      2552   Televisor  $122.59       7.0  11/06/2024        Electrónica
2091      1684  Aspiradora   $58.65       2.0  01/09/2024  Electrodomésticos
1528      2192   Televisor  $104.46      12.0  23/06/2024        Electrónica
862       1956    Secadora   $42.59      12.0  07/04/2024  Electrodomésticos
2542      1872    Heladera   $60.32       2.0  30/10/2024  Electrodomésticos

In [12]:
# Analizamos el campo precio
# Si es numericos se puede aplicar describe()
df_ventas["precio"].sample(5)

2639     $65.00
1150    $115.30
314      $65.00
729      $72.78
2657    $121.56
Name: precio, dtype: object

In [13]:
# Analizamos el campo cantidad
# si es numerico se puede aplicar describe()
df_ventas["cantidad"].sample(5)

2914    10.0
2915     8.0
1318    12.0
1605     1.0
3023     3.0
Name: cantidad, dtype: float64

In [14]:
# Analizamos el campo fecha_venta
df_ventas["fecha_venta"].sample(5)

257     31/01/2024
1467    13/06/2024
189     23/01/2024
3006    27/12/2024
2963    22/12/2024
Name: fecha_venta, dtype: object

In [15]:
# Analizamos el campo categoria
# Probar con value_counts y unique
df_ventas["categoria"].value_counts()

categoria
Decoración           1015
Electrónica          1012
Electrodomésticos    1008
Name: count, dtype: int64

### Calidad de datos
Actividad 5

In [16]:
# Buscamos registros duplicados
df_ventas.duplicated().sum()

np.int64(35)

In [17]:
# Visualizamos los duplicados
# keep: 'first', 'last', False
df_ventas[df_ventas.duplicated(keep='first')]

id_venta                producto   precio  cantidad fecha_venta  \
826        56                Cortinas   $66.24       5.0  05/04/2024   
827       421         Lámpara de mesa  $114.83       9.0  05/04/2024   
828       424       Jarrón decorativo   $87.94       2.0  05/04/2024   
829      1868                Cafetera   $62.23       1.0  05/04/2024   
830      2545             Auriculares   $32.81      11.0  05/04/2024   
831      2778              SmartWatch   $70.66       7.0  05/04/2024   
844       145                Alfombra   $41.72       8.0  06/04/2024   
845       300         Adorno de pared   $67.75       9.0  06/04/2024   
846       439         Lámpara de mesa   $35.43       2.0  06/04/2024   
847       906                Cortinas   $84.04       3.0  06/04/2024   
848       959         Lámpara de mesa   $76.32       7.0  06/04/2024   
849      1376                Secadora   $69.58       7.0  06/04/2024   
850      1776                Secadora   $33.55       3.0  06/04/2024   
851      1784              Aspiradora  $115.24       3.0  06/04/2024   
852      1974              Aspiradora   $87.32       7.0  06/04/2024   
853      2080  Consola de videojuegos  $100.69       7.0  06/04/2024   
854      2108              SmartWatch   $79.37      10.0  06/04/2024   
855      2169               Proyector  $120.34       1.0  06/04/2024   
867       213       Espejo decorativo   $40.76       8.0  07/04/2024   
868       345                Alfombra   $56.78       9.0  07/04/2024   
869       398       Rincón de plantas   $77.30       7.0  07/04/2024   
870       509         Lámpara de mesa   $72.50       1.0  07/04/2024   
871       821         Lámpara de mesa   $71.17      10.0  07/04/2024   
872      1642                Heladera   $88.78       2.0  07/04/2024   
873      1956                Secadora   $42.59      12.0  07/04/2024   
874      2085             Auriculares  $122.56       6.0  07/04/2024   
875      2096                  Laptop   $83.25       6.0  07/04/2024   
876      2356                  Laptop   $30.16       6.0  07/04/2024   
877      2373              Smartphone  $123.60      12.0  07/04/2024   
884       204       Jarrón decorativo   $94.75       1.0  08/04/2024   
885      1381      Freidora eléctrica   $38.12       2.0  08/04/2024   
886      2365             Auriculares   $92.91      11.0  08/04/2024   
887      2486                  Laptop  $124.95      11.0  08/04/2024   
888      2506                  Laptop   $34.74       1.0  08/04/2024   
889      2705             Auriculares   $87.29       2.0  08/04/2024   

             categoria  
826         Decoración  
827         Decoración  
828         Decoración  
829  Electrodomésticos  
830        Electrónica  
831        Electrónica  
844         Decoración  
845         Decoración  
846         Decoración  
847         Decoración  
848         Decoración  
849  Electrodomésticos  
850  Electrodomésticos  
851  Electrodomésticos  
852  Electrodomésticos  
853        Electrónica  
854        Electrónica  
855        Electrónica  
867         Decoración  
868         Decoración  
869         Decoración  
870         Decoración  
871         Decoración  
872  Electrodomésticos  
873  Electrodomésticos  
874        Electrónica  
875        Electrónica  
876        Electrónica  
877        Electrónica  
884         Decoración  
885  Electrodomésticos  
886        Electrónica  
887        Electrónica  
888        Electrónica  
889        Electrónica

In [18]:
# Buscamos datos nulos
df_ventas.isnull().sum()

id_venta       0
producto       0
precio         2
cantidad       2
fecha_venta    0
categoria      0
dtype: int64

In [19]:
# Visualizar los nulos
# df_ventas[df_ventas.isnull().any(axis=1)]
df_ventas[df_ventas.isnull().any(axis=1)]

id_venta               producto precio  cantidad fecha_venta    categoria
136       627  Elementos de cerámica    NaN       NaN  17/01/2024   Decoración
139      2171    Parlantes Bluetooth    NaN       NaN  17/01/2024  Electrónica

In [20]:
# Analizar los precios de los NaN a ver que opciones tenemos
# df_ventas[df_ventas["producto"]=="Elementos de cerámica"]
df_ventas["producto"].value_counts()
# Unique nos devuelve una lista de los valores unicos.
# df_ventas["producto"].unique()


producto
Lámpara de mesa           181
Auriculares               147
Microondas                135
Cafetera                  118
Laptop                    104
Secadora                  103
Jarrón decorativo         102
Smartphone                102
Alfombra                  102
SmartWatch                102
Aspiradora                102
Cortinas                  102
Rincón de plantas         102
Freidora eléctrica        101
Espejo decorativo         101
Adorno de pared           101
Lavadora                  100
Consola de videojuegos    100
Cuadro decorativo         100
Horno eléctrico           100
Plancha de vapor          100
Proyector                 100
Elementos de cerámica     100
Batidora                  100
Televisor                 100
Cámara digital            100
Parlantes Bluetooth        99
Tablet                     58
Heladera                   49
Candelabro                 24
Name: count, dtype: int64

### Observaciones




* Cantidad de registros: 3035
* Columnas 6
* 35 duplicados, eliminarlos
* Algunos nulos (2), 1% del faltante, dropna
* Campo precio: encontre `$`, removerlo y luego cambiar el dtype a float64
* Campos cantidad: cambiar dtype a int64
* Campo fecha: ajustar a dtype datetime

## Dataframe marketing

### Analisis exploratorio de datos

Aplicamos los métodos de EDA vistos en clase, `info`, `columns`, `dtypes`, `head`, `tail`, `sample`, `unique`, `value_counts`, `describe`, `duplicated`, `isnull`, etc. y documentamos las observaciones:

In [21]:
# Aplicamos info() para familiarizarnos con el dataframe
df_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id_campanha   90 non-null     int64  
 1   producto      90 non-null     object 
 2   canal         90 non-null     object 
 3   costo         90 non-null     float64
 4   fecha_inicio  90 non-null     object 
 5   fecha_fin     90 non-null     object 
dtypes: float64(1), int64(1), object(4)
memory usage: 4.3+ KB


In [22]:
# Visualizamos los primeros registros con head()
# también podemos usar tail() para visualizar los últimos
df_marketing.head()

id_campanha         producto  canal  costo fecha_inicio   fecha_fin
0           74  Adorno de pared     TV   4.81   20/03/2024  03/05/2024
1           12           Tablet   RRSS   3.40   26/03/2024  13/05/2024
2           32  Lámpara de mesa  Email   5.54   28/03/2024  20/04/2024
3           21       Smartphone   RRSS   6.37   29/03/2024  16/05/2024
4           58         Alfombra  Email   4.25   31/03/2024  05/05/2024

In [23]:
# Analizamos el campo costo
df_marketing["costo"].sample(5)

65    4.89
58    5.69
31    3.22
30    5.44
82    4.84
Name: costo, dtype: float64

In [24]:
# Analizamos el campo fecha_inicio
df_marketing["fecha_inicio"].sample(5)

33    10/07/2024
37    16/07/2024
87    17/12/2024
7     10/04/2024
40    24/07/2024
Name: fecha_inicio, dtype: object

In [25]:
# Analizamos el campo fecha_fin
df_marketing["fecha_fin"].sample(5)

85    29/12/2024
75    20/11/2024
7     08/06/2024
76    28/11/2024
2     20/04/2024
Name: fecha_fin, dtype: object

### Calidad de los datos


In [26]:
# Buscamos registros duplicados
df_marketing.duplicated().sum()

np.int64(0)

### Observaciones
* Cantidad de registro: 90
* Columnas: 6
* Duplicado: 0
* Fecha inicio y fin: Cambiar a dtype datetime

## Dataframe clientes

In [27]:
# Aplicamos info()
df_clientes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 567 entries, 0 to 566
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id_cliente  567 non-null    int64  
 1   nombre      567 non-null    object 
 2   edad        567 non-null    int64  
 3   ciudad      567 non-null    object 
 4   ingresos    567 non-null    float64
dtypes: float64(1), int64(2), object(2)
memory usage: 22.3+ KB


In [28]:
# Visualizamos con head()
df_clientes.head()

id_cliente               nombre  edad         ciudad  ingresos
0           1      Aloysia Screase    44  Mar del Plata  42294.68
1           2  Kristina Scaplehorn    25        Posadas  24735.04
2           3       Filip Castagne    50    Resistencia  35744.85
3           4          Liuka Luard    39   Bahía Blanca  27647.96
4           5        Dore Cockshtt    28        Rosario  28245.65

In [29]:
# Estadistica descriptiva
df_clientes.describe()

id_cliente        edad      ingresos
count  567.000000  567.000000    567.000000
mean   284.000000   37.940035  34668.739012
std    163.823075   10.202885  12974.531446
min      1.000000   20.000000    170.290000
25%    142.500000   30.000000  26015.240000
50%    284.000000   37.000000  35066.830000
75%    425.500000   43.000000  42457.100000
max    567.000000   81.000000  88053.010000

In [30]:
# Buscamos registros duplicados
df_clientes.duplicated().sum()

np.int64(0)

### Analisis exploratorio de datos

### Observaciones
* Cantidad de registros: 567
* Columnas: 5
* Duplicado: 0

# Etapa 2

## Limpieza de datos
Actividad 1 - Limpiar el conjunto de datos eliminando duplicados y caracteres no deseados. Documentar el proceso y los resultados.

### Dataframe ventas

* 35 duplicados, eliminarlos - se eliminaron 35 duplicados
* Algunos nulos (2), 1% del faltante, dropna - se eliminaron 2 registros con 2 NaN
* Cantidad de registros: 2998
* Campo precio: encontre `$`, removerlo y luego cambiar el dtype a float64 - Ok
* Campo cantidad: Cambiar el dtype a int64 - Ok
* Campo fecha: ajustar a dtype datetime - Ok

In [31]:
# Eliminamos registros duplicados
# atributos a evaluar, keep, ignore_index, inplace
df_ventas.drop_duplicates(inplace=True, ignore_index=True)

In [32]:
# Validamos que no haya duplicados
df_ventas.duplicated().sum()

np.int64(0)

In [33]:
# Eliminamos registros con datos nulos
df_ventas.dropna(inplace=True)

In [34]:
# Validamos que no haya nulos
# df_ventas.isnull().sum()
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2998 entries, 0 to 2999
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id_venta     2998 non-null   int64  
 1   producto     2998 non-null   object 
 2   precio       2998 non-null   object 
 3   cantidad     2998 non-null   float64
 4   fecha_venta  2998 non-null   object 
 5   categoria    2998 non-null   object 
dtypes: float64(1), int64(1), object(4)
memory usage: 164.0+ KB


In [35]:
# Eliminamos el caracter especial $ de precio
# Normalización: usar replace()
# De ser necesario, podemos usar el atributo regex=True
df_ventas["precio"] = df_ventas["precio"].str.strip().str.replace("$", "")

In [36]:
# Validamos que precio no contenga $
df_ventas["precio"].sample(3)

1865    41.28
1743    54.66
1210    61.76
Name: precio, dtype: object

In [37]:
# Convertimos precio de str a float
# Podemos usar pd.to_numeric o astype o incluso combinarlos
df_ventas["precio"] = df_ventas["precio"].astype(float)

In [38]:
# Convertir el dtype de cantidad, de float64 a int64
df_ventas["cantidad"]= df_ventas["cantidad"].astype(int)

In [39]:
# Validar la conversion a int
df_ventas["cantidad"].dtypes

dtype('int64')

In [40]:
# Convertir el campo fecha_venta a dtype datetime
# pd.to_datetime(df_ventas["fecha_venta"], format="%d/%m/%Y")
# pd.to_datetime(df_ventas["fecha_venta"], dayfirst=True)
df_ventas['fecha_venta'] = pd.to_datetime(df_ventas['fecha_venta'], format="%d/%m/%Y")

In [41]:
# Validar la conversion de fecha
df_ventas['fecha_venta'].dtype

dtype('<M8[ns]')

### Dataframe marketing

* Cantidad de registro: 90
* Fecha inicio: Cambiar a dtype datetime - Ok
* Fecha fin: Cambiar a dtype datetime - Ok

In [43]:
# Convertir el campo fecha_inicio a dtype datetime
df_marketing['fecha_inicio'] = pd.to_datetime(df_marketing['fecha_inicio'], format="%d/%m/%Y")

In [44]:
# Convertir el campo fecha_fin a dtype datetime
df_marketing['fecha_fin'] = pd.to_datetime(df_marketing['fecha_fin'], format="%d/%m/%Y")

In [47]:
# Validamos los cambios
df_marketing['fecha_inicio'].dtype
df_marketing['fecha_fin'].dtype


dtype('<M8[ns]')

## Transformación de datos
Actividad 2 - Aplicar filtros y transformaciones para crear una tabla de ventas que muestre solo los productos con alto rendimiento.

In [48]:
df_ventas.head(4)

id_venta           producto  precio  cantidad fecha_venta  \
0       792  Cuadro decorativo   69.94         5  2024-01-02   
1       811    Lámpara de mesa  105.10         5  2024-01-02   
2      1156           Secadora   97.96         3  2024-01-02   
3      1372           Heladera  114.35         8  2024-01-02   

           categoria  
0         Decoración  
1         Decoración  
2  Electrodomésticos  
3  Electrodomésticos

In [49]:
# Transformación:
# Calculamos el valor de cada venta = precio * cantidad
# Agregamos una nueva columnas
df_ventas["valor_venta"] = df_ventas["precio"] * df_ventas["cantidad"]

In [50]:
# Visualizamos
df_ventas.head(2)

id_venta           producto  precio  cantidad fecha_venta   categoria  \
0       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
1       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración   

   valor_venta  
0        349.7  
1        525.5

In [51]:
# Agregación:
# Agrupar valor_venta por producto
# Sumamos las ventas dentro de cada grupo
tabla_ventas = (df_ventas.groupby("producto", as_index=False)["valor_venta"].sum())
tabla_ventas.head()

producto  valor_venta
0  Adorno de pared     48093.49
1         Alfombra     44773.06
2       Aspiradora     50085.86
3      Auriculares     74175.58
4         Batidora     50979.20

In [52]:
# Usemos describe para analizar algunos valores del dataframe resultante
tabla_ventas.describe()

valor_venta
count     30.000000
mean   48903.117333
std    13265.561715
min    11128.800000
25%    45601.920000
50%    48140.245000
75%    51092.960000
max    82276.380000

## Agregación
Actividad 3 - Resumir las ventas por categoría de producto y analizar los ingresos generados.

In [55]:
df_ventas.columns

Index(['id_venta', 'producto', 'precio', 'cantidad', 'fecha_venta',
       'categoria', 'valor_venta'],
      dtype='object')

In [56]:
# Agrupamos por categoria y agregamos la suma
ventas_categoria = df_ventas.groupby("categoria", as_index=False)["valor_venta"].sum()
ventas_categoria.head()

categoria  valor_venta
0         Decoración    479216.09
1  Electrodomésticos    505299.63
2        Electrónica    482577.80

In [57]:
ventas_categoria

categoria  valor_venta
0         Decoración    479216.09
1  Electrodomésticos    505299.63
2        Electrónica    482577.80

In [58]:
# Ordenar y mostrar los resultados
ventas_categoria

categoria  valor_venta
0         Decoración    479216.09
1  Electrodomésticos    505299.63
2        Electrónica    482577.80

## Integración de datos
Actividad 4 - Combinar los sets de datos de ventas y marketing para obtener una visión más amplia de las tendencias.

In [59]:
df_ventas.head(2)

id_venta           producto  precio  cantidad fecha_venta   categoria  \
0       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
1       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración   

   valor_venta  
0        349.7  
1        525.5

In [60]:
df_marketing.head()

id_campanha         producto  canal  costo fecha_inicio  fecha_fin
0           74  Adorno de pared     TV   4.81   2024-03-20 2024-05-03
1           12           Tablet   RRSS   3.40   2024-03-26 2024-05-13
2           32  Lámpara de mesa  Email   5.54   2024-03-28 2024-04-20
3           21       Smartphone   RRSS   6.37   2024-03-29 2024-05-16
4           58         Alfombra  Email   4.25   2024-03-31 2024-05-05

In [61]:
# Unir los datasets ventas y marketing por columna 'producto'
ventas_marketing = pd.merge(df_ventas, df_marketing, on="producto", how="left")

In [62]:
# Verificar coincidencias (muestra)
ventas_marketing.head()

id_venta           producto  precio  cantidad fecha_venta   categoria  \
0       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
1       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
2       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
3       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración   
4       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración   

   valor_venta  id_campanha  canal  costo fecha_inicio  fecha_fin  
0        349.7            1   RRSS   5.27   2024-04-27 2024-06-04  
1        349.7           31  Email   5.28   2024-08-15 2024-09-12  
2        349.7           61     TV   5.30   2024-11-05 2024-12-23  
3        525.5           32  Email   5.54   2024-03-28 2024-04-20  
4        525.5            2   RRSS   5.88   2024-05-30 2024-06-29

### Analizamos el impacto de la campaña

In [66]:
# Nos quedamos solo con los registros de ventas durante la campaña de marketing
# Seleccionamos solo los campos que nos interesan: "producto", "valor_venta", "costo"
df_filtrado = ventas_marketing[(ventas_marketing["fecha_venta"] >= ventas_marketing["fecha_inicio"]) & (ventas_marketing["fecha_venta"] <= ventas_marketing["fecha_fin"])].copy()
df_resultado = df_filtrado[["producto", "valor_venta", "costo"]]

In [67]:
# Agrupamos por producto y hacemos una agregación por "valor de venta" y "costo"
ventas_marketing.groupby("categoria")[["valor_venta","costo"]].sum()

valor_venta     costo
categoria                               
Decoración          1437648.27  15774.86
Electrodomésticos   1515898.89  14644.65
Electrónica         1447733.40  14243.90

In [70]:
# Generamos una nueva columna con la diferencia entre lo vendido y lo invertido en la campaña
df_filtrado["ganancia"] = df_filtrado["valor_venta"] - df_filtrado["costo"]
# Verificamos
df_filtrado.head(10)

id_venta         producto  precio  cantidad fecha_venta    categoria  \
2073       950  Adorno de pared  116.42         6  2024-03-20   Decoración   
2091       780  Adorno de pared   79.13         7  2024-03-21   Decoración   
2124       270  Adorno de pared   96.59         6  2024-03-22   Decoración   
2160       210  Adorno de pared   71.67         4  2024-03-24   Decoración   
2256       730  Adorno de pared  103.90         8  2024-03-28   Decoración   
2286       749  Lámpara de mesa   44.88         1  2024-03-30   Decoración   
2298      2044           Tablet   35.84        11  2024-03-30  Electrónica   
2304      2683       Smartphone  111.41         7  2024-03-30  Electrónica   
2307      2984           Tablet   75.89        11  2024-03-30  Electrónica   
2313       221  Lámpara de mesa   41.81         9  2024-03-31   Decoración   

      valor_venta  id_campanha  canal  costo fecha_inicio  fecha_fin  ganancia  
2073       698.52           74     TV   4.81   2024-03-20 2024-05-03    693.71  
2091       553.91           74     TV   4.81   2024-03-20 2024-05-03    549.10  
2124       579.54           74     TV   4.81   2024-03-20 2024-05-03    574.73  
2160       286.68           74     TV   4.81   2024-03-20 2024-05-03    281.87  
2256       831.20           74     TV   4.81   2024-03-20 2024-05-03    826.39  
2286        44.88           32  Email   5.54   2024-03-28 2024-04-20     39.34  
2298       394.24           12   RRSS   3.40   2024-03-26 2024-05-13    390.84  
2304       779.87           21   RRSS   6.37   2024-03-29 2024-05-16    773.50  
2307       834.79           12   RRSS   3.40   2024-03-26 2024-05-13    831.39  
2313       376.29           32  Email   5.54   2024-03-28 2024-04-20    370.75